protoc-gen-swagger/genswagger/template.go

```diff
@@ -13,6 +13,49 @@ import (
 	"github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway/descriptor"
 )
 
+// messageToQueryParameters converts a message to a list of swagger query parameters.
+func messageToQueryParameters(message *descriptor.Message, reg *descriptor.Registry, pathParams []descriptor.Parameter) ([]swaggerParameterObject, error) {
+	var parameters []swaggerParameterObject
+	var addParameterWithPrefix func(string, *descriptor.Field) error
+	addParameterWithPrefix = func(prefix string, field *descriptor.Field) error {
+		// make sure the parameter is not already listed as a path parameter
+		for _, pathParam := range pathParams {
+			if pathParam.Target == field {
+				return nil
+			}
+		}
+		schema := schemaOfField(field, reg)
+		if schema.Type != "" {
+			// basic type, add a basic query parameter
+			parameters = append(parameters, swaggerParameterObject{
+				Name:        prefix + field.GetName(),
+				Description: schema.Description,
+				In:          "query",
+				Type:        schema.Type,
+			})
+			return nil
+		}
+
+		// nested type, recurse
+		msg, err := reg.LookupMsg("", field.GetTypeName())
+		if err != nil {
+			return fmt.Errorf("unknown message type %s", field.GetTypeName())
+		}
+		for _, nestedField := range msg.Fields {
+			if err := addParameterWithPrefix(field.GetName()+".", nestedField); err != nil {
```

Comment:
I'm also interested in the answer to this. What happens for deeply nested fields?